src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.Etw.cs

```diff
@@ -102,6 +102,10 @@ public void Test_EventSource_EtwManifestGenerationRollover()
 
                     tracesession.Flush();
 
+                    // Sleep after requesting flush to ensure that the manifest payload generated
+                    // is fully written to the etl file.
+                    Thread.Sleep(TimeSpan.FromSeconds(5));
```

Comment:
Any time I see sleeps being added to fend off race conditions that make tests fail, it raises red flags.  Presumably even with this sleep (and the others in the test) it's still possible for something to take an unexpectedly large amount of time and cause the test to fail?

Is there nothing else we can do? e.g. a polling loop that waits repeatedly checks for some state?  A retry on a particular known manifestation of this failure?  A callback notification to trigger the test to continue when everything is in an appropriate state?  Etc.

If we end up keeping these sleeps, at a minimum the test should be marked [OuterLoop].

Reply:
Thanks for the suggestion @stephentoub -- I'll fix this to add retry logic in these tests instead of putting arbitrary sleeps.

Reply:
Plus in my experience anything that normally takes ~1 sec can take up to 30 secs at least once when run enough zillion times in our test jobs, with busy machines, etc